docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala

```diff
@@ -27,4 +27,23 @@ class VerifyYamlTest extends FunSuite with Matchers {
 
     actualYamlNoIndent shouldBe expectedYaml
   }
+
+  val endpoint_wit_recursive_structure: Endpoint[(FruitAmount, String), Unit, (F1, Int)] = endpoint
+    .in(("fruit" / path[String] / "amount" / path[Int]).mapTo(FruitAmount))
```

Comment:
maybe we can keep this test simple and just test the recursive bit, removing the inputs and just using a single `F1` output